File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/abstractfactory/trasnport/Truck.java
package com.bigboxcode.designpattern.abstractfactory.trasnport;

public class Truck implements Transport {
    @Override
    public void start() {
        System.out.println("Truck Started");
    }

    @Override
    public void stop() {
        System.out.println("Truck Stopped");
    }

    @Override
    public void repair() {
        System.out.println("Truck Repair");
    }
}

File: design-patterns/ts/abstract-factory/transport/truck.ts
// truck.ts

import Transport from "./transport";


class Truck implements Transport {
    start(): void {
        console.log("Truck Started");
    }

    stop(): void {
        console.log("Truck Stopped");
    }

    repair(): void {
        console.log("Truck Repair");
    }
}

export default Truck;